### ISSUE TITLE
Assertion failure random crash when switching weapons or opening doors.

### ISSUE THREAD

**Issue by ZxSol** [nipkownix/re4_tweaks#564]
7I have the most recent version of RE4_tweaks installed along with the HD project, the game boots normally and lets me play for a while,, but after about 15 minutes or so, the game will crash throwing an assertion failure crash if I change weapons in the inventory menu when it leaves the menu, or when I enter a new room
https://imgur.com/gallery/ghgdTMk

**Comment by Almashina** [comment]
The same
![2023-06-01_190238](https://github.com/nipkownix/re4_tweaks/assets/54445863/1ef398e2-2d0b-4bcf-91a6-97bf2fd6db48)

**Comment by ZxSol** [comment]
little update, I switched to 1.8 version of RE4Tweaks and I havent gotten this error anymore, other people in the RE modding discord have also mentioned this too, so it seems its best to keep using 1.8 until this gets fixed

**Comment by nipkownix** [comment]
Please upload a CrashDump so we can take a look.

**Comment by Almashina** [comment]
Dear [nipkownix](https://github.com/nipkownix), how can we do this?
You don't mean **_game_output.log_**, do you?

**Comment by nipkownix** [comment]
>If your issue still persists feel free to make a report about it on the [GitHub issues page](https://github.com/nipkownix/re4_tweaks/issues) - if the problem is crashing/loading related you can try creating a CrashDumps folder inside the Bin32 directory (next to bio4.exe/steam_api.dll/etc), then the next time game crashes hopefully some info about it will be written in there, please include that with any reports you make!

**Comment by Mister-Curious** [comment]
I am also getting a lot of these, mostly  when retrying after ALT + Tabbing to desktop

**Comment by Almashina** [comment]
My _CrashDumps_ folder is still empty even after silent CTD.
Then I unchecked some of the options that were included in version 1.9 and after that there were no crashes in the game, 25-30 hours have already been played.
Unchecked options were:
**SpeedUpQuitGame**
**FixSilencedHandgunDescription**
If I'm not mistaken and this is not an accident, then this can be confirmed by other players who have CTD. Maybe we'll catch a bug.

**Comment by Mister-Curious** [comment]
@Almashina Yes, this is because this error doesn't actually trigger a crash in the same way. We can often ignore this pop-up warning and continue to play by pressing 'Ignore'. This being said, it is still a big problem to have this message pop up.

I will check the 'SpeedUpQuitGame' option first and see if that helps. 

**EDIT -** Still got the problem happening with the 'SpeedUpQuitGame' option disabled

**Comment by DarthxVoid** [comment]
@nipkownix in my mod I use the option of speed and size of the Enemies a lot, and it gives this direct error even though it does not give anything in the game, the people who test my mod were afraid to continue the game and they closed it
And this error with me never results in a crash

**Comment by Almashina** [comment]
[re4_tweaks.zip](https://github.com/nipkownix/re4_tweaks/files/11711808/re4_tweaks.zip)
[dinput8.zip](https://github.com/nipkownix/re4_tweaks/files/11711959/dinput8.zip)
This is my re4_tweaks folder and dinput8.ini with dinput8.log files with no CTD or Visual C++ errors for about 25-30 hours to date.
The game setup is: 
- ordinary steam re4 hd edition 1.1.0
- hd project 1.1
- re4_tweaks 1.9.1
- no additional mods/changes
- my own ingame and re4_tweak settings

It would be nice if someone could check all this config for crashes and/or c++ errors.
________________________________________________________
[Mister-Curious](https://github.com/Mister-Curious), I was getting:
- CTD with c++ error
- CTD without any pop-ups - "silent" CTD
- C++ error without crash, which can be ignored.

In the second case, I have a "silent" CTD - the CrashDumps folder was left empty. After that, I changed some settings, unfortunately, I didn’t mark exactly which ones at that time (except SpeedUpQuitGame and FixSilencedHandgunDescription), but the errors and crashes stopped.
I can't check the CrashDumps folder for the 1st and 3rd cases because there are no more crashes/errors to this point.

**Comment by Mister-Curious** [comment]
In hopes of getting this cleared up I removed all instances of scaled enemies in the room I was working on, yet the problem persists. The lack of CrashDumps for this obviously makes it harder to troubleshoot, so I am at a loss as for what is causing this .

**Comment by Almashina** [comment]
Hi all!
I caught a crash to desktop after many dozens of hours of bug-free play. Not sure if this is due to a C++ bug, but luckily there is a dump.
The moment in the game is mission 5-3, Mike the helicopter guy appearance level. This crash occurred when leaving the level and using the gate. The error doesn't happen all the time, but I've encountered it before in the same place, possibly on "separate ways".
[bio4.exe.20230625001137.zip](https://github.com/nipkownix/re4_tweaks/files/11859656/bio4.exe.20230625001137.zip)
